include/circt/Dialect/SV/Expressions.td

```diff
@@ -22,3 +22,28 @@ def TextualValueOp : SVOp<"textual_value", [NoSideEffect]> {
      $string attr-dict `:` type($result)
   }];
 }
+
+// A list of things that the extract op knows with how to deal.
+def Extractable : Type<Or<[
+  CPred<"$_self.isa<::circt::sv::InterfaceType>()">
+]>, "Types from which ports or values can be extracted.">;
+
+def ExtractOp : SVOp<"extract", [NoSideEffect]> {
+  let summary = "SystemVerilog '.' operator";
+  let description = [{
+    The SystemVerilog '.' operator can be used to access
+    fields/members/modports in types (structs or unions) or interfaces. It is
+    not the '.' used in instantiations to connect signals (`.clk(clk)`).
+
+    Get the 'dataflow_in' modport on the 'handshake_example' interface:
+    ```
+    %ifaceModport = sv.extract @dataflow_in %iface : !sv.interface<@handshake_example>
+    ```
+  }];
+
+  let arguments = (ins Extractable:$extractable, FlatSymbolRefAttr:$field);
+  let results = (outs AnyType:$result);
```

Comment:
Could this be a predicate for `InterfaceModport` types instead of `AnyType`?

Reply:
I went with `AnyType` to allow for future extraction of structs. Though we could broaden it later on. Maybe it's not appropriate to use the same op for both...? I did it this way in the spirit of the SV dialect being an "AST" dialect.

Reply:
Right, yes that makes sense. I think if this op is going to be general with different types accepted for `Extractable`, then definitely the result type should be open. I guess I'll amend my comment to say maybe it can be a list of allowable types, like `Extractable`. I guess `AnyType` could be fine here too.